src/main/java/com/fasterxml/jackson/databind/deser/MissingInstantiatorHandler.java

```diff
@@ -0,0 +1,41 @@
+package com.fasterxml.jackson.databind.deser;
+
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.ValueDeserializer;
+import com.fasterxml.jackson.databind.util.ReflectionUtil;
+
+public class MissingInstantiatorHandler extends DeserializationProblemHandler {
+
+    private static volatile MissingInstantiatorHandler INSTANCE;
+
+    private MissingInstantiatorHandler() {
+    }
+
+    public static MissingInstantiatorHandler getInstance() {
+        if (INSTANCE == null) {
+            synchronized (MissingInstantiatorHandler.class) {
+                if (INSTANCE == null) {
+                    INSTANCE = new MissingInstantiatorHandler();
+                }
+            }
+        }
+        return INSTANCE;
+    }
+
+    @Override
+    public Object handleMissingInstantiator(DeserializationContext ctxt, Class<?> instClass, ValueInstantiator valueInsta,
+                                            JsonParser jsonParser, String msg) throws JacksonException {
+        Object instance = ReflectionUtil.newConstructorAndCreateInstance(instClass);
+        if (instance == null) {
+            return NOT_HANDLED;
+        }
+
+        ValueDeserializer<Object> deserializer = ctxt.findRootValueDeserializer(ctxt.constructType(instClass));
```

Comment:
Hmm. I am not confident this would for general cases wrt, say, polymorphic handling or with any generic types (`instClass` is type erased). While this would probably work for POJOs I am bit concerned about this being the wrong place to handle this issue. At the same time I realize that there may not be a good alternative yet for providing "alternate" no-arguments constructor yet; it would need to be added to be accessed via `ValueInstantiator`

I guess I'll have to think about this part a bit more.